ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -1320,6 +1320,18 @@ KEXT_STATIC bool CurrentProcessWasSpawnedByRegularUser()
         proc_rele(process);
     }
     
+    if (!nonServiceUser)
+    {
+       // We need to allow the system process amfid to run
+       // It is required for 'open project.app'
+       char buffer[MAXCOMLEN + 1] = "";
+       proc_selfname(buffer, sizeof(buffer));
+       if (0 == strcmp(buffer, "amfid"))
```

Comment:
For my own curiosity, as someone not familiar with the code :) - why do we operate with c styled strings, instead of the string type? The proc name is already stored as a string, and we have to convert it to a c style string. Would it be possible to stay in the domain of the String type?

Reply:
> why do we operate with c styled strings, instead of the string type?  The proc name is already stored as a string, and we have to convert it to a c style string.

This is because STL is not supported for kernel extensions.  The spot you saw where the proc name is a string is in the unit tests (where we are allowed to use STL).

Reply:
ah! thanks!

Reply:
We don't have the 'string' type in KauthHandler.cpp.  This is on the 'hot' path for the kext so we want to stay as lean and fast as possible.  The test code does have 'string' for ease of use.